core/field_number.js

```diff
@@ -157,13 +157,14 @@ Blockly.FieldNumber.prototype.setConstraints_ = function(opt_min, opt_max,
  * Show the inline free-text editor on top of the text and the num-pad if
  * appropriate.
  * @param {!Event} e A mouse down or touch start event.
+ * @param {boolean=} opt_showNumPad If true, show the num pad.
  * @private
  */
-Blockly.FieldNumber.prototype.showEditor_ = function(e) {
+Blockly.FieldNumber.prototype.showEditor_ = function(e, opt_showNumPad) {
   Blockly.FieldNumber.activeField_ = this;
   // Do not focus on mobile devices so we can show the num-pad
-  var showNumPad =
-      goog.userAgent.MOBILE || goog.userAgent.ANDROID || goog.userAgent.IPAD;
+  var showNumPad = opt_showNumPad ||
```

Comment:
This change allows you to force the numpad to show, but not force it to **not** show (i.e. if you set the flag to false, it will still be shown on mobile)

If you want both force hide and force show, you'll need 2 separate flags

Reply:
Too true..